server/auth/auth_handler.go

```diff
@@ -97,6 +97,9 @@ func (h *AuthorizationHandler) HandleAuth(ctx *sql.Context, aqs sql.Authorizatio
 
 	var privileges []Privilege
 	switch auth.AuthType {
+	case AuthType_IGNORE:
```

Comment:
Should add an auth test for CTEs that makes sure you can't access a table using a CTE as a loophole

Reply:
Turns out auth doesn't work for CTEs, but it doesn't work in GMS either.
Added a skipped test.
Trackig issue here: https://github.com/dolthub/go-mysql-server/issues/2746